### PR TITLE
- [waf] skip test for cross-building (m1)

### PR DESCRIPTION
**Pull request by SSE4**
Specify library name and version:  **waf/all**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/waf/all/test_package/conanfile.py [hunk 1]
```diff
@@ -9,11 +9,16 @@ class TestPackageConan(ConanFile):
     exports_sources = "a.cpp", "b.cpp", "main.c", "main.cpp", "wscript"
```
Comment (madebr) [inline recipes/waf/all/test_package/conanfile.py:10]: Perhaps this should contain `test_type = "build_requires"` + `required_conan_version = ">=1.36.0"`?

https://docs.conan.io/en/latest/devtools/build_requires.html#testing-build-requires

Comment (SSE4) [reply recipes/waf/all/test_package/conanfile.py:10]: it may contain that, still, `waf.exe` will be ARMv8 and it will fail?

Comment (madebr) [reply recipes/waf/all/test_package/conanfile.py:10]: `waf` is python-only. There is no binary that I'm aware of.
This recipe still uses `os_build` and `arch_build`.
Perhaps switching to `os`/`arch` will improve cross builds?

Comment (SSE4) [reply recipes/waf/all/test_package/conanfile.py:10]: it should, I am not a fan of `os_build`/`arch_build`

**Review by jgsogo — CHANGES_REQUESTED** [review]
M1 profiles doesn't provide `os/arch_build`, so we really need to remove them from `settings`

https://github.com/conan-io/conan-center-index/blob/03815a5c9f6d084bf5f641be3759d68dd9676fc5/recipes/waf/all/conanfile.py#L12

**Review by jgsogo — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**